public/javascripts/SVValidate/src/mission/Mission.js

```diff
@@ -103,6 +105,33 @@ function Mission(params) {
         svv.statusField.setProgressText(completionRate);
     }
 
+    /**
+     * Updates status bar (UI) and current mission properties whenever undo is pressed.
+     */
+    function updateMissionProgressUndo() {
+        setProperty("labelsProgress", getProperty("labelsProgress") - 1);
+        setProperty("labelsValidated", getProperty("labelsValidated"));
+        let labelsProgress = getProperty("labelsProgress");
+        let completionRate = labelsProgress / getProperty("labelsValidated");
+        svv.statusField.setProgressBar(completionRate);
+        svv.statusField.setProgressText(completionRate);
+        if (lastStatus === 1) {
```

Comment:
Instead of keeping track of how you the user validated their previous label in `Mission.js`, can we just get that value from something like one of these places?
```
svv.panorama.getCurrentLabel().getProperty('validationResult')
```
or
```
svv.panorama.getLastLabel().validation_result
```

On a related note, when a label is validated, the Label object has their `validationResult` property set. But you're not turning it back to `undefined` when the validation is undone.

As I look through the code a bit more, do you think it would make more sense to kick off all the undo-related stuff from in `Label.js` so that it mirrors `validate()` in `Label.js`? Thankfully it will be more simple than the validate function, but it does seem weird that Label.js isn't involved in this process when it's the primary driver of doing the validation itself!